internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -262,6 +271,32 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 				}, false),
 		},
 
+		"nginx_app_protect": {
```

Comment:
Does the API allow the user to specify WAF if plan/SKU is `basic`?

Reply:
basic plan doesn't support WAF so NGINX RP will reject the request

Reply:
Thanks for confirming. As long as the service API handles it 👍